usaspending_api/agency/v2/views/agency_base.py

```diff
@@ -44,3 +51,26 @@ def toptier_agency(self):
     @property
     def standard_response_messages(self):
         return [get_account_data_time_period_message()] if self.fiscal_year < 2017 else []
+
+    @cached_property
+    def pagination(self):
+        if self.request.method == "GET":
+            return None
+        sortable_columns = ["name", "obligated_amount", "gross_outlay_amount"]
+        default_sort_column = "obligated_amount"
+        model = customize_pagination_with_sort_columns(sortable_columns, default_sort_column)
+        request_data = TinyShield(model).block(self.request.data)
+        return Pagination(
+            page=request_data["page"],
+            limit=request_data["limit"],
+            lower_limit=(request_data["page"] - 1) * request_data["limit"],
+            upper_limit=(request_data["page"] * request_data["limit"]) + 1,
+            sort_key=request_data.get("sort", "obligated_amount"),
+            sort_order=request_data["order"],
+        )
+
+    @property
+    def filter(self):
```

Comment:
I'm a little on the fence about adding this (and `pagination`) to the base class since it is not ubiquitous to most of the endpoints.  I know both of these are dirty words, but we could subclass this specifically for endpoints that provide pagination and filtering or we could create a mixin that supports that functionality.

A third option would be to support this compositionally (which is a lot like subclassing and/or mixins except your class uses the base class rather than actually inheriting from it).  Composition is so IN right now.  You want to be cool, don't you?